### DISCUSSION TITLE
Codecept should not use exit below zero nor above 127.

### DISCUSSION THREAD

**Discussion by imme-emosol** [Codeception/Codeception#6330] [Ideas]
To allow `git bisect run` to "work", prevent codecept from exiting with a state that git bisect run does not recognize.

> [...] exit with a code between 1 and 127 (inclusive), except 125, if the current source code is bad/new.
> Any other exit code will abort the bisect process. It should be noted that a program that terminates via exit(-1) leaves $? = 255 [...]

~ https://git-scm.com/docs/git-bisect#_bisect_run .

An example of where this leads to an issue, is:
https://github.com/Codeception/Codeception/blob/f9bca68b330806b2ab28203c9b8ccb2c5f21cae1/src/Codeception/Subscriber/ErrorHandler.php#L128

Introduced in https://github.com/Codeception/Codeception/commit/d5333a5e5ea4f5511cf9859a7c5adfcf128417a1

**Comment by Naktibalda** [comment]
I think that it would make sense to use exit code 125.

> The special exit code 125 should be used when the current source code cannot be tested.

**Reply by imme-emosol** [reply]
~~yes.. i wrote differently in my pr, didn't i? :)~~
fixed

**Comment by imme-emosol** [comment]
The idea of this discussion is to perhaps think about ways to write or change the tests for code additions so those tests can prevent exit-states that are outside the range that git bisect run "knows".
